server/cluster/cluster.go

```diff
@@ -144,7 +145,11 @@ func (s *Server) getCluster(ctx context.Context, q *cluster.ClusterQuery) (*appv
 		q.Server = ""
 		q.Name = ""
 		if q.Id.Type == "name" {
-			q.Name = q.Id.Value
+			name, err := url.QueryUnescape(q.Id.Value)
```

Comment:
This would be a breaking API change for all clusters for which `url.QueryUnescape(name) != name`. For example, if my cluster were named (for some weird reason) `in%40cluster`, I'd now get a 403 when requesting that cluster _unless_ I did so with the UI.

Could we introduce a new `name_escaped` to solve the bug and still support other use cases?

Reply:
Got it, I changed the UI to submit with type `name_escaped` instead and adjusted the backend side of things to handle that appropriately while still preserving the original behavior. I also duplicated the test to check that clusters where something in their name could be interpreted as escaped will pass through unmodified.